client/network/src/transport.rs

```diff
@@ -121,11 +137,23 @@ pub fn build_transport(
 		core::upgrade::SelectUpgrade::new(yamux_config, mplex_config)
 	};
 
-	let transport = transport
+	let tcp_transport = transport
 		.upgrade(upgrade::Version::V1Lazy)
 		.authenticate(authentication_config)
 		.multiplex(multiplexing_config)
-		.timeout(Duration::from_secs(20))
+		.timeout(Duration::from_secs(20));
+
+	let quic_transport = if enable_quic {
+		OptionalTransport::some(build_quic_transport(&keypair))
+	} else {
+		OptionalTransport::none()
+	};
+
+	let transport = OrTransport::new(quic_transport, tcp_transport)
```

Comment:
```suggestion
	let transport = OrTransport::new(tcp_transport, quic_transport)
```

I think `tcp_transport` should be preferable (i.e. tried first)